includes/backend.inc

```diff
@@ -1022,7 +1022,8 @@ function _drush_backend_invoke($cmds, $common_backend_options = array(), $contex
     $ret = array();
     foreach ($procs as $site => $proc) {
       if (($proc['code'] == DRUSH_APPLICATION_ERROR) && isset($common_backend_options['integrate'])) {
-        drush_set_error('DRUSH_APPLICATION_ERROR', dt("The external command could not be executed due to an application error."));
+        drush_set_error('DRUSH_APPLICATION_ERROR', dt("The external command could not be executed due to an application error. Try running the command again with the --debug option."));
+        drush_log(dt('The raw command and output is !proc', array('!proc' => var_export($proc, TRUE))), 'debug');
```

Comment:
1. $proc can be giant, with log messages, return value, errors, and output. I think this should be $proc['output']? Paging @greg-1-anderson.
1. Furthermore, there is a line later which attempts to log the output. Would be good to understand why that didn't help you. See `$ret = drush_set_error('DRUSH_FRAMEWORK_ERROR', dt("The command could not be executed successfully (returned: !return, code: !code)", array("!return" => $proc['output'], "!code" =>  $proc['code'])));`